### PR TITLE
Test with Junit

### PR DESCRIPTION
**Pull request by josehpena**
I add a junit test at this commit.

### CODE REVIEW

File: challenge/src/main/java/com/ifood/challenge/location/LocationController.java [hunk 1]
```diff
@@ -5,20 +5,28 @@
...
+
     @GetMapping
+    public List homePage(){
+        return List.of("Hello World");
+    }
+    
+    @GetMapping(path="/{number}")
     public List<Location> getLocations(@PathVariable("number")int number){
-        return locationService.getLocation(number);
+        return List.of(locationService.getLocation(number));
```
Comment (ratoji) [inline challenge/src/main/java/com/ifood/challenge/location/LocationController.java:29]: Why did you get rid of `locationService`? 🤔

File: challenge/src/main/java/com/ifood/challenge/location/LocationService.java [hunk 2]
```diff
@@ -2,23 +2,23 @@
...
-import java.util.List;
-import java.util.Random;
-
 @Service
 public class LocationService {
-    public List<Location> getLocation(int number){
+    public Location[] getLocation(int number){
         Location locations[] = new Location[number];
+        double doubleFirst = 1.000;
+        double doubleSecond = 2.000;
```
Comment (ratoji) [inline challenge/src/main/java/com/ifood/challenge/location/LocationService.java:10]: The exercise states that random numbers should be generated:

https://rwondemand.atlassian.net/browse/DRVDXP-177

File: challenge/src/test/java/com/ifood/challenge/ChallengeApplicationTest.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package com.ifood.challenge;
+
+import com.ifood.challenge.location.Location;
+import com.ifood.challenge.location.LocationService;
+import org.junit.Before;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class ChallengeApplicationTest extends LocationService {
```
Comment (ratoji) [inline challenge/src/test/java/com/ifood/challenge/ChallengeApplicationTest.java:10]: A test is always independent from any other system resources...

So, you cannot extend `LocationService` here.
What you're saying is: my test is a location service. Does it makes sense?

The exercise required an integration test, which is a very specific kind of test.

So I suggest you understand some boundaries like:

- Unit test
- Integration test

And some resources you're gonna need to properly execute integration tests on Spring:

- `@SpringBootTest` annotation
- `@Autowire` to provide dependency injection into your test. Knowing what dependency injection is, is a good start also

File: challenge/src/test/java/com/ifood/challenge/ChallengeApplicationTest.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    Location[] arrayForTest = new Location[passInt];
+
+
+    @Test
+    public void testResult(){
+        arrayForTest[0] = new Location("Location 0", 1.0, 2.0);
+        arrayForTest[1] = new Location("Location 1", 1.001, 2.001);
+        Location[] locations = getLocation(passInt);
+
+        assertArrayEquals(arrayForTest, locations);
```
Comment (ratoji) [inline challenge/src/test/java/com/ifood/challenge/ChallengeApplicationTest.java:23]: So... Here you're testing that your API always works for some `passInt`. Does it works for any `passInt`?

Test scenarios suggestions:

- Happy paths: call for 1, maybe 10
- Corner cases: what about 0? -1? 

Try to follow something like: https://martinfowler.com/bliki/GivenWhenThen.html

File: challenge/src/main/java/com/ifood/challenge/location/LocationController.java [hunk 5]
```diff
@@ -5,20 +5,28 @@
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
+import javax.swing.plaf.synth.SynthMenuBarUI;
```
Comment (ratoji) [inline challenge/src/main/java/com/ifood/challenge/location/LocationController.java:8]: It doesn't look like it belongs here...

**Review by ratoji — CHANGES_REQUESTED** [review]
Suggestion: create specifc tasks to handle and fix the comments below.

Make sure you understand what you're doing and try to ask for help in a well-structured manner.

File: challenge/src/main/java/com/ifood/challenge/location/LocationController.java (hunk 1 above)
Comment (josehpena) [reply challenge/src/main/java/com/ifood/challenge/location/LocationController.java:29]: I just change the getLocation to return the array, before that it returned a "List.of".

File: challenge/src/main/java/com/ifood/challenge/location/LocationService.java (hunk 2 above)
Comment (josehpena) [reply challenge/src/main/java/com/ifood/challenge/location/LocationService.java:10]: I had forgotten that when I did the unit test, i change back

File: challenge/src/test/java/com/ifood/challenge/ChallengeApplicationTest.java (hunk 3 above)
Comment (ratoji) [reply challenge/src/test/java/com/ifood/challenge/ChallengeApplicationTest.java:10]: You can check `@MockMvc` also.
